migration/migration/015-tracing-compression.sql

```diff
@@ -261,82 +261,135 @@ GRANT SELECT, INSERT, UPDATE, DELETE ON TABLE _ps_trace.link TO prom_writer;
-                number_partitions=>1::int,
-                chunk_time_interval=>'1 hours'::interval,
-                create_default_indexes=>false
-            );
-            PERFORM public.create_distributed_hypertable(
-                '_ps_trace.link'::regclass,
-                'span_start_time'::name,
-                partitioning_column=>'trace_id'::name,
-                number_partitions=>1::int,
-                chunk_time_interval=>'1 hours'::interval,
-                create_default_indexes=>false
-            );
-        ELSE -- not multinode
-            PERFORM public.create_hypertable(
-                '_ps_trace.span'::regclass,
-                'start_time'::name,
-                partitioning_column=>'trace_id'::name,
-                number_partitions=>1::int,
-                chunk_time_interval=>'1 hours'::interval,
-                create_default_indexes=>false
-            );
-            PERFORM public.create_hypertable(
-                '_ps_trace.event'::regclass,
-                'time'::name,
-                partitioning_column=>'trace_id'::name,
-                number_partitions=>1::int,
-                chunk_time_interval=>'1 hours'::interval,
-                create_default_indexes=>false
-            );
-            PERFORM public.create_hypertable(
-                '_ps_trace.link'::regclass,
-                'span_start_time'::name,
-                partitioning_column=>'trace_id'::name,
-                number_partitions=>1::int,
-                chunk_time_interval=>'1 hours'::interval,
-                create_default_indexes=>false
-            );
-        END IF;
-
-        IF _is_compression_available THEN
-            -- turn on compression
-            ALTER TABLE _ps_trace.span SET (timescaledb.compress, timescaledb.compress_orderby='trace_id,span_id,start_time');
-            ALTER TABLE _ps_trace.event SET (timescaledb.compress, timescaledb.compress_orderby='trace_id,span_id,time');
-            ALTER TABLE _ps_trace.link SET (timescaledb.compress, timescaledb.compress_orderby='trace_id,span_id,span_start_time');
-
-            PERFORM public.add_compression_policy('_ps_trace.span', INTERVAL '1 hours');
-            PERFORM public.add_compression_policy('_ps_trace.event', INTERVAL '1 hours');
-            PERFORM public.add_compression_policy('_ps_trace.link', INTERVAL '1 hours');
-        END IF;
+    PERFORM public.set_chunk_time_interval(
+        '_ps_trace.span'::regclass,
+        '1 hour'::interval
+    );
+    PERFORM public.set_chunk_time_interval(
+        '_ps_trace.event'::regclass,
+        '1 hour'::interval
+    );
+    PERFORM public.set_chunk_time_interval(
+        '_ps_trace.link'::regclass,
+        '1 hour'::interval
+    );
+
+    _is_compression_available = NOT (current_setting('timescaledb.license') = 'apache');
+    IF _is_compression_available THEN
+        -- drop any compressed chunks for the three tables we are modifying
```

Comment:
I think this will go away after rebase; It should be around, we are truncating the tables on top now.